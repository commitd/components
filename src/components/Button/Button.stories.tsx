import React from 'react'
import { Story, Meta } from '@storybook/react'
import { Button } from './Button'
import { Variants } from '../../docs/util'
import { Row, Grid } from '../'

export default {
  title: 'Components/Button',
  component: Button,
  argTypes: {
    onClick: { action: 'clicked' },
    /**
     * react-docgen-typescript fails us here.
     * @see https://github.com/storybookjs/storybook/issues/12641#issuecomment-708603353
     */
    as: {
      control: {
        type: 'select',
        options: ['button', 'a'],
      },
      defaultValue: 'button',
      description:
        'Can be any HTML element, but can also be any component. Typically only used with "button" or "a".',
    },
  },
} as Meta

export const Default = () => {
  return <Button>Button</Button>
}

const Template: Story = (args) => <Button {...args} />

export const Primary = Template.bind({})
Primary.args = { children: 'Button' }

export const Size = () => (
  <Row css={{ gap: '$3' }}>
    <Button size="small">Small</Button>
    <Button size="default">Default</Button>
    <Button size="large">Large</Button>
  </Row>
)

export const Variant = () => (
  <Row css={{ gap: '$3' }}>
    <Button variant="primary">Primary</Button>
    <Button variant="secondary">Secondary</Button>
    <Button variant="tertiary">Tertiary</Button>
  </Row>
)

export const Destructive = () => (
  <Row css={{ gap: '$3' }}>
    <Button destructive variant="primary">
      Primary
    </Button>
    <Button destructive variant="secondary">
      Secondary
    </Button>
    <Button destructive variant="tertiary">
      Tertiary
    </Button>
  </Row>
)

export const Disabled = () => (
  <Row css={{ gap: '$3' }}>
    <Button disabled variant="primary">
      Primary
    </Button>
    <Button disabled variant="secondary">
      Secondary
    </Button>
    <Button disabled variant="tertiary">
      Tertiary
    </Button>
  </Row>
)

/**
 * A `brand` variant is also supplied for particular uses cases, like in the AppBar, where changing to the dark theme does not affect the coloring.
 */
export const Brand = () => <Button variant="brand">Brand</Button>

/**
 * This uses the force prop to simulate hover, focus and active states so they can be compared at the same time.
 * This prop is not intended for normal use and the buttons here will not interact normally.
 */
export const State = () => (
  <Grid css={{ gap: '$3', gridTemplateColumns: '1fr 1fr 1fr 1fr' }}>
    <Button variant="primary">Button</Button>
    <Button variant="primary" force="hover">
      Button
    </Button>
    <Button variant="primary" force="focus">
      Button
    </Button>
    <Button variant="primary" force="active">
      Button
    </Button>
    <Button variant="secondary">Button</Button>
    <Button variant="secondary" force="hover">
      Button
    </Button>
    <Button variant="secondary" force="focus">
      Button
    </Button>
    <Button variant="secondary" force="active">
      Button
    </Button>
    <Button variant="tertiary">Button</Button>
    <Button variant="tertiary" force="hover">
      Button
    </Button>
    <Button variant="tertiary" force="focus">
      Button
    </Button>
    <Button variant="tertiary" force="active">
      Button
    </Button>
  </Grid>
)

export const All = () => (
  <Variants
    component={Button}
    variant={['primary', 'secondary', 'tertiary']}
    size={['small', 'default', 'large']}
    destructive={[false, true]}
    disabled={[false, true]}
    children="Button"
  />
)
