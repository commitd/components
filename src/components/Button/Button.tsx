import React from 'react'
import { StitchesVariants, styled } from 'stitches.config'

const DEFAULT_TAG = 'button'

export const hover = {
  background:
    '$$hover radial-gradient(circle, transparent 1%, $$hover 1%) center/15000%',
}

export const focus = hover

export const active = {
  backgroundColor: '$$active',
  backgroundSize: '100%',
  transition: 'background 0s',
}

export const mainVariants = {
  brand: {
    $$active: '$colors$brandActive',
    $$lowlight: '$colors$brandLowlight',
    $$hover: '$colors$brandHighlight',
    backgroundColor: '$colors$brand',
    color: '$colors$brandContrast',
  },
  primary: {
    $$active: '$colors$primaryActive',
    $$lowlight: '$colors$primaryLowlight',
    $$hover: '$$mainHover',
    backgroundColor: '$$main',
    color: '$$contrast',
  },
  secondary: {
    borderColor: '$$default',
    border: 'solid 2px',
    color: '$$default',
    $$hover: '$$defaultHover',
  },
  tertiary: {
    color: '$$default',
    $$hover: '$$defaultHover',
  },
}

export const destructiveVariants = {
  false: {},
  true: {
    $$main: '$colors$error',
    $$mainHover: '$colors$errorHighlight',
    $$contrast: '$colors$errorContrast',
    $$active: '$colors$errorActive',
    $$default: '$colors$error',
    $$defaultHover: '$colors$errorBackground',
    $$lowlight: '$colors$errorLowlight',
  },
}

export const sizeVariants = {
  small: {
    paddingLeft: '$3',
    paddingRight: '$3',
    height: '$5',
    fontSize: '$-1',
  },
  default: {
    paddingLeft: '$4',
    paddingRight: '$4',
    height: '$6',
  },
  large: {
    paddingLeft: '$6',
    paddingRight: '$6',
    height: '$7',
  },
}

export const buttonBaseStyle = {
  $$main: '$colors$primary',
  $$mainHover: '$colors$primaryHighlight',
  $$contrast: '$colors$primaryContrast',
  $$active: '$colors$defaultActive',
  $$default: '$colors$default',
  $$defaultHover: '$colors$defaultHighlight',
  $$lowlight: '$colors$defaultLowlight',

  // Reset
  alignItems: 'center',
  appearance: 'none',
  boxSizing: 'border-box',
  display: 'inline-flex',
  flexShrink: 0,
  justifyContent: 'center',
  lineHeight: '$none',
  margin: '0',
  outline: 'none',
  padding: '0',
  textDecoration: 'none',
  userSelect: 'none',
  WebkitTapHighlightColor: 'rgba(0,0,0,0)',
  '&::before': {
    boxSizing: 'border-box',
  },
  '&::after': {
    boxSizing: 'border-box',
  },

  // Defaults
  fontSize: '$0',
  borderRadius: '$default',
  cursor: 'pointer',
  backgroundColor: 'transparent',
  border: 'none',
  width: 'fit-content',

  // Actions
  transition: 'background 0.5s',
  backgroundPosition: 'center',

  '&:hover': hover,
  '&:focus': focus,
  '&:active': active,

  '&:disabled': {
    pointerEvents: 'none',
    $$main: '$$lowlight',
    $$default: '$$lowlight',
  },
}

const buttonVariants = {
  variant: mainVariants,
  destructive: destructiveVariants,
  size: sizeVariants,
  force: {
    hover,
    focus,
    active,
  },
}

const StyledButton = styled(DEFAULT_TAG, {
  ...buttonBaseStyle,
  variants: buttonVariants,
  defaultVariants: {
    variant: 'secondary',
    destructive: 'false',
    size: 'default',
  },
} as Parameters<typeof styled>[1] & {
  variants: typeof buttonVariants
})

type ButtonVariants = StitchesVariants<typeof StyledButton>
type ButtonOwnProps = React.ComponentProps<typeof StyledButton>
export type ButtonProps = ButtonOwnProps & {
  /**  This is a test of the docs process */
  variant?: ButtonVariants['variant']
}

/**
 * Button component
 */
export const Button: React.FC<ButtonProps> = StyledButton
Button.toString = () => `.${StyledButton.className}`
