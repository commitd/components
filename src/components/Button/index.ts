export { Button,buttonBaseStyle } from './Button'
