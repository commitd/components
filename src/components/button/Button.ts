import * as React from "react"
import styled from "styled-components"
import { width, space, WidthProps, SpaceProps } from "styled-system"
import theme, { ThemeProps } from "../../theme/theme"

export type ButtonProps = WidthProps &
  SpaceProps &
  ThemeProps & {
    // Size variations of the button
    size: "small" | "medium" | "large"
    // Set true to disable
    disabled?: boolean
  }

const size = (props: ButtonProps) => {
  switch (props.size) {
    case "small":
      return {
        fontSize: `${props.theme!.fontSizes[0]}px`,
        padding: "7px 12px"
      }
    case "medium":
      return {
        fontSize: `${props.theme!.fontSizes[1]}px`,
        padding: "9.5px 18px"
      }
    case "large":
      return {
        fontSize: `${props.theme!.fontSizes[2]}px`,
        padding: "12px 22px"
      }
    default:
      return {
        fontSize: `${props.theme!.fontSizes[1]}px`,
        padding: "9.5px 18px"
      }
  }
}

export const Button: React.FC<ButtonProps> = styled.button<ButtonProps>`
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  text-decoration: none;
  font-family: inherit;
  font-weight: 600;
  line-height: 1.5;
  cursor: pointer;
  border-radius: ${props => props.theme.radius};
  background-color: ${props => props.theme.colors.blue};
  color: ${props => props.theme.colors.white};
  border-width: 0;
  border-style: solid;
  &:disabled {
    opacity: 0.25;
  }
  &:hover {
    background-color: ${props =>
      props.disabled ? null : props.theme.colors.darkBlue};
  }
  ${width} ${size} ${space};
`

Button.defaultProps = {
  theme: theme
}

Button.displayName = "Button"

export default Button
