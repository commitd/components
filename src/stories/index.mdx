import { Meta } from '@storybook/addon-docs/blocks'
import { version } from '../../package.json'

<Meta title="Design System|Introduction" />

# Committed Components

<code>v{version}</code>

This is Committed's design system and should be used to compose web applications.

<!-- ## Sketch Kit

<ButtonLink href='https://github.com/pricelinelabs/design-system-sketch/releases/latest'>
  Download Sketch Kit
</ButtonLink> -->

## Motivation

Design systems enable teams to build better products faster by making design reusable—reusability makes scale possible. This is the heart and primary value of design systems. A design system is a collection of reusable components, guided by clear standards, that can be assembled together to build any number of applications.

Regardless of the technologies and tools behind them, a successful design system follows these guiding principles:

- **It’s consistent**. The way components are built and managed follows a predictable pattern.
- **It’s self-contained**. Your design system is treated as a standalone dependency.
- **It’s reusable**. You’ve built components so they can be reused in many contexts.
- **It’s accessible**. Applications built with your design system are usable by as many people as possible, no matter how they access the web.
- **It’s robust**. No matter the product or platform to which your design system is applied, it should perform with grace and minimal bugs.

## Next

- [Getting Started](gettingStarted)
