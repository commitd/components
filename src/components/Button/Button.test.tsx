import { composeStories } from '@storybook/react'
import React from 'react'
import { renderDark, renderLight, screen, userEvent } from '../../test'
import * as stories from './Button.stories'

const { Default, Disabled, All } = composeStories(stories)

it('renders and can be clicked', () => {
  const onClickSpy = jest.fn()
  renderLight(<Default onClick={onClickSpy} />)
  const buttonElement = screen.getByRole('button')
  buttonElement.click()
  expect(onClickSpy).toHaveBeenCalled()
})

it('renders and can be accessed with keyboard', () => {
  const onClickSpy = jest.fn()
  renderLight(<Default onClick={onClickSpy} />)
  userEvent.tab()
  userEvent.keyboard('{enter}')
  expect(onClickSpy).toHaveBeenCalled()
})

it('renders disabled without role', async () => {
  renderDark(<Disabled />)
  const buttons = await screen.findAllByRole('button')
  buttons.forEach((b) => expect(b).toHaveAttribute('disabled'))
  // NOTE should throw due to pointerEvents: 'none' but still allowed to force so this doesn't work.
  // expect(() => userEvent.click(buttons[0])).toThrow()
})

it('renders all variants without error', () => {
  const { asFragment } = renderDark(<All />)
  expect(asFragment()).toBeDefined()
})
