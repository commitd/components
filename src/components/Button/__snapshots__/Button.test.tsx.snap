// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders dark without error 1`] = `
<DocumentFragment>
  :root,
.sxfmafz {
  --sx-colors-yellow100: #ffeecc;
  --sx-colors-yellow200: #ffde99;
  --sx-colors-yellow300: #ffcd66;
  --sx-colors-yellow400: #ffbd33;
  --sx-colors-yellow500: #ffac00;
  --sx-colors-yellow600: #cc8a00;
  --sx-colors-yellow700: #996700;
  --sx-colors-yellow800: #664500;
  --sx-colors-yellow900: #332200;
  --sx-colors-grey100: #d8d8d8;
  --sx-colors-grey200: #b2b2b2;
  --sx-colors-grey300: #8b8b8b;
  --sx-colors-grey400: #656565;
  --sx-colors-grey500: #3e3e3e;
  --sx-colors-grey600: #323232;
  --sx-colors-grey700: #252525;
  --sx-colors-grey800: #191919;
  --sx-colors-grey900: #0c0c0c;
  --sx-colors-blue100: #d4e6f3;
  --sx-colors-blue200: #a8cce7;
  --sx-colors-blue300: #7db3da;
  --sx-colors-blue400: #5199ce;
  --sx-colors-blue500: #2680c2;
  --sx-colors-blue600: #1e669b;
  --sx-colors-blue700: #174d74;
  --sx-colors-blue800: #0f334e;
  --sx-colors-blue900: #081a27;
  --sx-colors-green100: #d4eee6;
  --sx-colors-green200: #a9ddcd;
  --sx-colors-green300: #7dcdb5;
  --sx-colors-green400: #52bc9c;
  --sx-colors-green500: #27ab83;
  --sx-colors-green600: #1f8969;
  --sx-colors-green700: #17674f;
  --sx-colors-green800: #104434;
  --sx-colors-green900: #08221a;
  --sx-colors-red100: #f1d3d3;
  --sx-colors-red200: #e3a8a8;
  --sx-colors-red300: #d67c7c;
  --sx-colors-red400: #c85151;
  --sx-colors-red500: #ba2525;
  --sx-colors-red600: #951e1e;
  --sx-colors-red700: #701616;
  --sx-colors-red800: #4a0f0f;
  --sx-colors-red900: #250707;
  --sx-colors-orange100: #fae4d7;
  --sx-colors-orange200: #f5c8ae;
  --sx-colors-orange300: #f0ad86;
  --sx-colors-orange400: #eb915d;
  --sx-colors-orange500: #e67635;
  --sx-colors-orange600: #b85e2a;
  --sx-colors-orange700: #8a4720;
  --sx-colors-orange800: #5c2f15;
  --sx-colors-orange900: #2e180b;
  --sx-colors-brandYellow: #ffbb00;
  --sx-colors-brandGrey: #565555;
  --sx-colors-background: #f7f7f7;
  --sx-colors-text: #3b3b3b;
  --sx-colors-primary: var(--sx-colors-grey500);
  --sx-colors-secondary: var(--sx-colors-yellow200);
  --sx-colors-paper: #ffffff;
  --sx-space-0: 0px;
  --sx-space-1: 4px;
  --sx-space-2: 8px;
  --sx-space-3: 12px;
  --sx-space-4: 16px;
  --sx-space-5: 24px;
  --sx-space-6: 32px;
  --sx-space-7: 48px;
  --sx-space-8: 64px;
  --sx-space-9: 96px;
  --sx-space-10: 128px;
  --sx-space-11: 192px;
  --sx-space-12: 256px;
  --sx-space-13: 384px;
  --sx-space-14: 512px;
  --sx-space-15: 640px;
  --sx-space-16: 768px;
  --sx-fontSizes-0: 16px;
  --sx-fontSizes-1: 18px;
  --sx-fontSizes-2: 20px;
  --sx-fontSizes-3: 24px;
  --sx-fontSizes-4: 30px;
  --sx-fontSizes-5: 36px;
  --sx-fontSizes-6: 48px;
  --sx-fontSizes-7: 60px;
  --sx-fontSizes-8: 72px;
  --sx-fontSizes--2: 12px;
  --sx-fontSizes--1: 14px;
  --sx-fonts-system: -apple-system, BlinkMacSystemFont, "San Francisco", Roboto,  "Segoe UI", "Helvetica Neue";
  --sx-fonts-display: "Helvetica Neue", "Segoe UI", Helvetica, Arial, sans-serif;
  --sx-fonts-monospace: "SFMono-Regular", Consolas, "Liberation Mono", "Andale Mono", "Ubuntu Mono", Menlo, Courier, monospace;
  --sx-fonts-text: "Inter" -apple-system, BlinkMacSystemFont, "San Francisco", Roboto,  "Segoe UI", "Helvetica Neue";
  --sx-fontWeights-regular: 400;
  --sx-fontWeights-bold: 700;
  --sx-fontWeights-light: 300;
  --sx-lineHeights-body: 140%;
  --sx-lineHeights-auto: AUTO;
  --sx-lineHeights-none: 100%;
  --sx-lineHeights-tight: 110%;
  --sx-letterSpacings-normal: 0%;
  --sx-letterSpacings-tight: -1%;
  --sx-letterSpacings-tighter: -3%;
  --sx-letterSpacings-wide: 5%;
  --sx-borderWidths-default: 2px;
  --sx-radii-default: 2px;
  --sx-radii-round: 99999px;
}

.dark-theme {
  --sx-colors-background: #000000;
  --sx-colors-text: #f7f7f7;
  --sx-colors-primary: var(--sx-colors-yellow500);
  --sx-colors-secondary: var(--sx-colors-616161);
  --sx-colors-paper: #222222;
}

@global {
  text-size-adjust: 100%;
  text-rendering: optimizeLegibility;
}

html {
  overflow-x: hidden;
}

body {
  margin: 0;
}

* {
  margin: 0;
  padding: 0;
}

svg {
  display: block;
}

pre,
code {
  margin: 0;
  font-family: var(--sx-fonts-monospace);
}

body,
button {
  font-family: var(--sx-fonts-text);
}

.sx8b8d3 {
  background-color: gainsboro;
  border-radius: 9999px;
  font-size: 13px;
  border: 0;
}

.sx8b8d3qj8w3--color-gray {
  background-color: gainsboro;
}

.sx8b8d3qj8w3--color-gray:hover {
  background-color: lightgray;
}

:root,
.sxfmafz {
  --sx-colors-yellow100: #ffeecc;
  --sx-colors-yellow200: #ffde99;
  --sx-colors-yellow300: #ffcd66;
  --sx-colors-yellow400: #ffbd33;
  --sx-colors-yellow500: #ffac00;
  --sx-colors-yellow600: #cc8a00;
  --sx-colors-yellow700: #996700;
  --sx-colors-yellow800: #664500;
  --sx-colors-yellow900: #332200;
  --sx-colors-grey100: #d8d8d8;
  --sx-colors-grey200: #b2b2b2;
  --sx-colors-grey300: #8b8b8b;
  --sx-colors-grey400: #656565;
  --sx-colors-grey500: #3e3e3e;
  --sx-colors-grey600: #323232;
  --sx-colors-grey700: #252525;
  --sx-colors-grey800: #191919;
  --sx-colors-grey900: #0c0c0c;
  --sx-colors-blue100: #d4e6f3;
  --sx-colors-blue200: #a8cce7;
  --sx-colors-blue300: #7db3da;
  --sx-colors-blue400: #5199ce;
  --sx-colors-blue500: #2680c2;
  --sx-colors-blue600: #1e669b;
  --sx-colors-blue700: #174d74;
  --sx-colors-blue800: #0f334e;
  --sx-colors-blue900: #081a27;
  --sx-colors-green100: #d4eee6;
  --sx-colors-green200: #a9ddcd;
  --sx-colors-green300: #7dcdb5;
  --sx-colors-green400: #52bc9c;
  --sx-colors-green500: #27ab83;
  --sx-colors-green600: #1f8969;
  --sx-colors-green700: #17674f;
  --sx-colors-green800: #104434;
  --sx-colors-green900: #08221a;
  --sx-colors-red100: #f1d3d3;
  --sx-colors-red200: #e3a8a8;
  --sx-colors-red300: #d67c7c;
  --sx-colors-red400: #c85151;
  --sx-colors-red500: #ba2525;
  --sx-colors-red600: #951e1e;
  --sx-colors-red700: #701616;
  --sx-colors-red800: #4a0f0f;
  --sx-colors-red900: #250707;
  --sx-colors-orange100: #fae4d7;
  --sx-colors-orange200: #f5c8ae;
  --sx-colors-orange300: #f0ad86;
  --sx-colors-orange400: #eb915d;
  --sx-colors-orange500: #e67635;
  --sx-colors-orange600: #b85e2a;
  --sx-colors-orange700: #8a4720;
  --sx-colors-orange800: #5c2f15;
  --sx-colors-orange900: #2e180b;
  --sx-colors-brandYellow: #ffbb00;
  --sx-colors-brandGrey: #565555;
  --sx-colors-background: #f7f7f7;
  --sx-colors-text: #3b3b3b;
  --sx-colors-primary: var(--sx-colors-grey500);
  --sx-colors-secondary: var(--sx-colors-yellow200);
  --sx-colors-paper: #ffffff;
  --sx-space-0: 0px;
  --sx-space-1: 4px;
  --sx-space-2: 8px;
  --sx-space-3: 12px;
  --sx-space-4: 16px;
  --sx-space-5: 24px;
  --sx-space-6: 32px;
  --sx-space-7: 48px;
  --sx-space-8: 64px;
  --sx-space-9: 96px;
  --sx-space-10: 128px;
  --sx-space-11: 192px;
  --sx-space-12: 256px;
  --sx-space-13: 384px;
  --sx-space-14: 512px;
  --sx-space-15: 640px;
  --sx-space-16: 768px;
  --sx-fontSizes-0: 16px;
  --sx-fontSizes-1: 18px;
  --sx-fontSizes-2: 20px;
  --sx-fontSizes-3: 24px;
  --sx-fontSizes-4: 30px;
  --sx-fontSizes-5: 36px;
  --sx-fontSizes-6: 48px;
  --sx-fontSizes-7: 60px;
  --sx-fontSizes-8: 72px;
  --sx-fontSizes--2: 12px;
  --sx-fontSizes--1: 14px;
  --sx-fonts-system: -apple-system, BlinkMacSystemFont, "San Francisco", Roboto,  "Segoe UI", "Helvetica Neue";
  --sx-fonts-display: "Helvetica Neue", "Segoe UI", Helvetica, Arial, sans-serif;
  --sx-fonts-monospace: "SFMono-Regular", Consolas, "Liberation Mono", "Andale Mono", "Ubuntu Mono", Menlo, Courier, monospace;
  --sx-fonts-text: "Inter" -apple-system, BlinkMacSystemFont, "San Francisco", Roboto,  "Segoe UI", "Helvetica Neue";
  --sx-fontWeights-regular: 400;
  --sx-fontWeights-bold: 700;
  --sx-fontWeights-light: 300;
  --sx-lineHeights-body: 140%;
  --sx-lineHeights-auto: AUTO;
  --sx-lineHeights-none: 100%;
  --sx-lineHeights-tight: 110%;
  --sx-letterSpacings-normal: 0%;
  --sx-letterSpacings-tight: -1%;
  --sx-letterSpacings-tighter: -3%;
  --sx-letterSpacings-wide: 5%;
  --sx-borderWidths-default: 2px;
  --sx-radii-default: 2px;
  --sx-radii-round: 99999px;
}

<div
    class="dark-theme"
  >
    <button
      class="sx8b8d3 sx8b8d3qj8w3--color-gray"
    />
  </div>
</DocumentFragment>
`;

exports[`renders light without error 1`] = `
<DocumentFragment>
  :root,
.sxfmafz {
  --sx-colors-yellow100: #ffeecc;
  --sx-colors-yellow200: #ffde99;
  --sx-colors-yellow300: #ffcd66;
  --sx-colors-yellow400: #ffbd33;
  --sx-colors-yellow500: #ffac00;
  --sx-colors-yellow600: #cc8a00;
  --sx-colors-yellow700: #996700;
  --sx-colors-yellow800: #664500;
  --sx-colors-yellow900: #332200;
  --sx-colors-grey100: #d8d8d8;
  --sx-colors-grey200: #b2b2b2;
  --sx-colors-grey300: #8b8b8b;
  --sx-colors-grey400: #656565;
  --sx-colors-grey500: #3e3e3e;
  --sx-colors-grey600: #323232;
  --sx-colors-grey700: #252525;
  --sx-colors-grey800: #191919;
  --sx-colors-grey900: #0c0c0c;
  --sx-colors-blue100: #d4e6f3;
  --sx-colors-blue200: #a8cce7;
  --sx-colors-blue300: #7db3da;
  --sx-colors-blue400: #5199ce;
  --sx-colors-blue500: #2680c2;
  --sx-colors-blue600: #1e669b;
  --sx-colors-blue700: #174d74;
  --sx-colors-blue800: #0f334e;
  --sx-colors-blue900: #081a27;
  --sx-colors-green100: #d4eee6;
  --sx-colors-green200: #a9ddcd;
  --sx-colors-green300: #7dcdb5;
  --sx-colors-green400: #52bc9c;
  --sx-colors-green500: #27ab83;
  --sx-colors-green600: #1f8969;
  --sx-colors-green700: #17674f;
  --sx-colors-green800: #104434;
  --sx-colors-green900: #08221a;
  --sx-colors-red100: #f1d3d3;
  --sx-colors-red200: #e3a8a8;
  --sx-colors-red300: #d67c7c;
  --sx-colors-red400: #c85151;
  --sx-colors-red500: #ba2525;
  --sx-colors-red600: #951e1e;
  --sx-colors-red700: #701616;
  --sx-colors-red800: #4a0f0f;
  --sx-colors-red900: #250707;
  --sx-colors-orange100: #fae4d7;
  --sx-colors-orange200: #f5c8ae;
  --sx-colors-orange300: #f0ad86;
  --sx-colors-orange400: #eb915d;
  --sx-colors-orange500: #e67635;
  --sx-colors-orange600: #b85e2a;
  --sx-colors-orange700: #8a4720;
  --sx-colors-orange800: #5c2f15;
  --sx-colors-orange900: #2e180b;
  --sx-colors-brandYellow: #ffbb00;
  --sx-colors-brandGrey: #565555;
  --sx-colors-background: #f7f7f7;
  --sx-colors-text: #3b3b3b;
  --sx-colors-primary: var(--sx-colors-grey500);
  --sx-colors-secondary: var(--sx-colors-yellow200);
  --sx-colors-paper: #ffffff;
  --sx-space-0: 0px;
  --sx-space-1: 4px;
  --sx-space-2: 8px;
  --sx-space-3: 12px;
  --sx-space-4: 16px;
  --sx-space-5: 24px;
  --sx-space-6: 32px;
  --sx-space-7: 48px;
  --sx-space-8: 64px;
  --sx-space-9: 96px;
  --sx-space-10: 128px;
  --sx-space-11: 192px;
  --sx-space-12: 256px;
  --sx-space-13: 384px;
  --sx-space-14: 512px;
  --sx-space-15: 640px;
  --sx-space-16: 768px;
  --sx-fontSizes-0: 16px;
  --sx-fontSizes-1: 18px;
  --sx-fontSizes-2: 20px;
  --sx-fontSizes-3: 24px;
  --sx-fontSizes-4: 30px;
  --sx-fontSizes-5: 36px;
  --sx-fontSizes-6: 48px;
  --sx-fontSizes-7: 60px;
  --sx-fontSizes-8: 72px;
  --sx-fontSizes--2: 12px;
  --sx-fontSizes--1: 14px;
  --sx-fonts-system: -apple-system, BlinkMacSystemFont, "San Francisco", Roboto,  "Segoe UI", "Helvetica Neue";
  --sx-fonts-display: "Helvetica Neue", "Segoe UI", Helvetica, Arial, sans-serif;
  --sx-fonts-monospace: "SFMono-Regular", Consolas, "Liberation Mono", "Andale Mono", "Ubuntu Mono", Menlo, Courier, monospace;
  --sx-fonts-text: "Inter" -apple-system, BlinkMacSystemFont, "San Francisco", Roboto,  "Segoe UI", "Helvetica Neue";
  --sx-fontWeights-regular: 400;
  --sx-fontWeights-bold: 700;
  --sx-fontWeights-light: 300;
  --sx-lineHeights-body: 140%;
  --sx-lineHeights-auto: AUTO;
  --sx-lineHeights-none: 100%;
  --sx-lineHeights-tight: 110%;
  --sx-letterSpacings-normal: 0%;
  --sx-letterSpacings-tight: -1%;
  --sx-letterSpacings-tighter: -3%;
  --sx-letterSpacings-wide: 5%;
  --sx-borderWidths-default: 2px;
  --sx-radii-default: 2px;
  --sx-radii-round: 99999px;
}

@global {
  text-size-adjust: 100%;
  text-rendering: optimizeLegibility;
}

html {
  overflow-x: hidden;
}

body {
  margin: 0;
}

* {
  margin: 0;
  padding: 0;
}

svg {
  display: block;
}

pre,
code {
  margin: 0;
  font-family: var(--sx-fonts-monospace);
}

body,
button {
  font-family: var(--sx-fonts-text);
}

.sx8b8d3 {
  background-color: gainsboro;
  border-radius: 9999px;
  font-size: 13px;
  border: 0;
}

.sx8b8d3qj8w3--color-gray {
  background-color: gainsboro;
}

.sx8b8d3qj8w3--color-gray:hover {
  background-color: lightgray;
}

:root,
.sxfmafz {
  --sx-colors-yellow100: #ffeecc;
  --sx-colors-yellow200: #ffde99;
  --sx-colors-yellow300: #ffcd66;
  --sx-colors-yellow400: #ffbd33;
  --sx-colors-yellow500: #ffac00;
  --sx-colors-yellow600: #cc8a00;
  --sx-colors-yellow700: #996700;
  --sx-colors-yellow800: #664500;
  --sx-colors-yellow900: #332200;
  --sx-colors-grey100: #d8d8d8;
  --sx-colors-grey200: #b2b2b2;
  --sx-colors-grey300: #8b8b8b;
  --sx-colors-grey400: #656565;
  --sx-colors-grey500: #3e3e3e;
  --sx-colors-grey600: #323232;
  --sx-colors-grey700: #252525;
  --sx-colors-grey800: #191919;
  --sx-colors-grey900: #0c0c0c;
  --sx-colors-blue100: #d4e6f3;
  --sx-colors-blue200: #a8cce7;
  --sx-colors-blue300: #7db3da;
  --sx-colors-blue400: #5199ce;
  --sx-colors-blue500: #2680c2;
  --sx-colors-blue600: #1e669b;
  --sx-colors-blue700: #174d74;
  --sx-colors-blue800: #0f334e;
  --sx-colors-blue900: #081a27;
  --sx-colors-green100: #d4eee6;
  --sx-colors-green200: #a9ddcd;
  --sx-colors-green300: #7dcdb5;
  --sx-colors-green400: #52bc9c;
  --sx-colors-green500: #27ab83;
  --sx-colors-green600: #1f8969;
  --sx-colors-green700: #17674f;
  --sx-colors-green800: #104434;
  --sx-colors-green900: #08221a;
  --sx-colors-red100: #f1d3d3;
  --sx-colors-red200: #e3a8a8;
  --sx-colors-red300: #d67c7c;
  --sx-colors-red400: #c85151;
  --sx-colors-red500: #ba2525;
  --sx-colors-red600: #951e1e;
  --sx-colors-red700: #701616;
  --sx-colors-red800: #4a0f0f;
  --sx-colors-red900: #250707;
  --sx-colors-orange100: #fae4d7;
  --sx-colors-orange200: #f5c8ae;
  --sx-colors-orange300: #f0ad86;
  --sx-colors-orange400: #eb915d;
  --sx-colors-orange500: #e67635;
  --sx-colors-orange600: #b85e2a;
  --sx-colors-orange700: #8a4720;
  --sx-colors-orange800: #5c2f15;
  --sx-colors-orange900: #2e180b;
  --sx-colors-brandYellow: #ffbb00;
  --sx-colors-brandGrey: #565555;
  --sx-colors-background: #f7f7f7;
  --sx-colors-text: #3b3b3b;
  --sx-colors-primary: var(--sx-colors-grey500);
  --sx-colors-secondary: var(--sx-colors-yellow200);
  --sx-colors-paper: #ffffff;
  --sx-space-0: 0px;
  --sx-space-1: 4px;
  --sx-space-2: 8px;
  --sx-space-3: 12px;
  --sx-space-4: 16px;
  --sx-space-5: 24px;
  --sx-space-6: 32px;
  --sx-space-7: 48px;
  --sx-space-8: 64px;
  --sx-space-9: 96px;
  --sx-space-10: 128px;
  --sx-space-11: 192px;
  --sx-space-12: 256px;
  --sx-space-13: 384px;
  --sx-space-14: 512px;
  --sx-space-15: 640px;
  --sx-space-16: 768px;
  --sx-fontSizes-0: 16px;
  --sx-fontSizes-1: 18px;
  --sx-fontSizes-2: 20px;
  --sx-fontSizes-3: 24px;
  --sx-fontSizes-4: 30px;
  --sx-fontSizes-5: 36px;
  --sx-fontSizes-6: 48px;
  --sx-fontSizes-7: 60px;
  --sx-fontSizes-8: 72px;
  --sx-fontSizes--2: 12px;
  --sx-fontSizes--1: 14px;
  --sx-fonts-system: -apple-system, BlinkMacSystemFont, "San Francisco", Roboto,  "Segoe UI", "Helvetica Neue";
  --sx-fonts-display: "Helvetica Neue", "Segoe UI", Helvetica, Arial, sans-serif;
  --sx-fonts-monospace: "SFMono-Regular", Consolas, "Liberation Mono", "Andale Mono", "Ubuntu Mono", Menlo, Courier, monospace;
  --sx-fonts-text: "Inter" -apple-system, BlinkMacSystemFont, "San Francisco", Roboto,  "Segoe UI", "Helvetica Neue";
  --sx-fontWeights-regular: 400;
  --sx-fontWeights-bold: 700;
  --sx-fontWeights-light: 300;
  --sx-lineHeights-body: 140%;
  --sx-lineHeights-auto: AUTO;
  --sx-lineHeights-none: 100%;
  --sx-lineHeights-tight: 110%;
  --sx-letterSpacings-normal: 0%;
  --sx-letterSpacings-tight: -1%;
  --sx-letterSpacings-tighter: -3%;
  --sx-letterSpacings-wide: 5%;
  --sx-borderWidths-default: 2px;
  --sx-radii-default: 2px;
  --sx-radii-round: 99999px;
}

<div
    class="light-theme"
  >
    <button
      class="sx8b8d3 sx8b8d3qj8w3--color-gray"
    />
  </div>
</DocumentFragment>
`;
